cloud/services/routetables/routetables.go

```diff
@@ -25,78 +25,69 @@ import (
-		s.Scope.ControlPlaneSubnet().RouteTable.ID = to.String(existingRouteTable.ID)
+			// route table already exists
+			// currently don't support:
+			//  1. creating separate control plane and node (#718) so update both
+			//  2. specifying your own routes via spec
+			s.Scope.NodeSubnet().RouteTable.Name = to.String(existingRouteTable.Name)
+			s.Scope.NodeSubnet().RouteTable.ID = to.String(existingRouteTable.ID)
+			s.Scope.ControlPlaneSubnet().RouteTable.Name = to.String(existingRouteTable.Name)
+			s.Scope.ControlPlaneSubnet().RouteTable.ID = to.String(existingRouteTable.ID)
 
-		return nil
-	}
+			return nil
+		}
 
-	s.Scope.V(2).Info("creating route table", "route table", routeTableSpec.Name)
-	err = s.Client.CreateOrUpdate(
-		ctx,
-		s.Scope.ResourceGroup(),
-		routeTableSpec.Name,
-		network.RouteTable{
-			Location:                   to.StringPtr(s.Scope.Location()),
-			RouteTablePropertiesFormat: &network.RouteTablePropertiesFormat{},
-		},
-	)
-	if err != nil {
-		return errors.Wrapf(err, "failed to create route table %s in resource group %s", routeTableSpec.Name, s.Scope.ResourceGroup())
-	}
+		s.Scope.V(2).Info("creating route table", "route table", rtSpec.Name)
+		err = s.Client.CreateOrUpdate(
+			ctx,
+			s.Scope.ResourceGroup(),
+			rtSpec.Name,
+			network.RouteTable{
+				Location:                   to.StringPtr(s.Scope.Location()),
+				RouteTablePropertiesFormat: &network.RouteTablePropertiesFormat{},
+			},
+		)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create route table %s in resource group %s", rtSpec.Name, s.Scope.ResourceGroup())
+		}
 
-	s.Scope.V(2).Info("successfully created route table", "route table", routeTableSpec.Name)
+		s.Scope.V(2).Info("successfully created route table", "route table", rtSpec.Name)
+	}
 	return nil
 }
 
 // Delete deletes the route table with the provided name.
-func (s *Service) Delete(ctx context.Context, spec interface{}) error {
+func (s *Service) Delete(ctx context.Context) error {
 	if !s.Scope.Vnet().IsManaged(s.Scope.ClusterName()) {
 		s.Scope.V(4).Info("Skipping route table deletion in custom vnet mode")
 		return nil
 	}
-	routeTableSpec, ok := spec.(*Spec)
-	if !ok {
-		return errors.New("invalid Route Table Specification")
-	}
-	s.Scope.V(2).Info("deleting route table", "route table", routeTableSpec.Name)
-	err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), routeTableSpec.Name)
-	if err != nil && azure.ResourceNotFound(err) {
-		// already deleted
-		return nil
-	}
-	if err != nil {
-		return errors.Wrapf(err, "failed to delete route table %s in resource group %s", routeTableSpec.Name, s.Scope.ResourceGroup())
-	}
+	for _, rtSpec := range s.Scope.RouteTableSpecs() {
+		s.Scope.V(2).Info("deleting route table", "route table", rtSpec.Name)
+		err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), rtSpec.Name)
+		if err != nil && azure.ResourceNotFound(err) {
+			// already deleted
+			return nil
```

Comment:
this needs to `continue` instead so it deletes the other route tables in the list

Reply:
fixed